specification.markdown

```diff
@@ -1,66 +1,140 @@
+    size beforehand.
+
+- Often very space efficient, even compared to other binary
+    representations of the same data.
+
+Some of the EBML disadvantages are:
+
+- No references can be made between EBML files, such as includes or
+    inherits. Every EBML document is a self contained entity. The data
+    stored in EBML may of course reference other resources.
+
+- No compositioning process to merge two or more EBML languages
+    currently exists.
+
+This document describes the EBML binary syntax, its semantic
+interpretation and the syntax of a textual document type definition
+format used to define the rules and constraints of an EBML language.
+
+## Notations and Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in RFC 2119.
+
+BNF is used throughout this document to augment the description and
+make it more formalized. It must however be noted that two different
+formats are described in this document, with two different BNFs. One
+for the binary format in chapter 2 and one for the document type
+definition in the rest of the document. To avoid confusion different
+token names has been chosen. The BNFs can be viewed in full in the
+appendices.
+
+## EBML principle
+
+EBML specifies a binary and octet (byte) aligned format inspired by the
+principle of XML. EBML itself is a generalized description of the
+technique of binary markup. Like XML, it is completely agnostic to any
+data that it might contain. The format is made of 2 parts: the semantic
+and the syntax. The semantic specifies a number of Element IDs and their basic type and is not included in the data file/stream.
+
+Just like XML, the specific "tags" (Element IDs in EBML parlance) used in an EBML implementation are arbitrary. However, the semantic of EBML
+outlines general data types and Element IDs.
 
 The known basic types are:
 
--   Signed Integer - Big-endian, any size from 1 to 8 octets
--   Unsigned Integer - Big-endian, any size from 1 to 8 octets
--   Float - Big-endian, defined for 4 and 8 octets (32, 64 bits)
--   String - Printable ASCII (0x20 to 0x7E), zero-padded when needed
--   UTF-8 - [Unicode string](http://www.unicode.org/), zero padded when
-    needed ([RFC 2279](http://www.faqs.org/rfcs/rfc2279.html))
--   Date - signed 8 octets integer in nanoseconds with 0 indicating the
-    precise beginning of the millennium (at
-    2001-01-01T00:00:00,000000000 UTC)
--   master-element - contains other EBML sub-elements of the next lower
-    level
--   Binary - not interpreted by the parser
+Data Type        | Definition
+-----------------|----------------------------------------------------
+Signed Integer   | Big-endian, any size from 1 to 8 octets
+Unsigned Integer | Big-endian, any size from 1 to 8 octets
+Float            | Big-endian, defined for 4 and 8 octets (32, 64
```

Comment:
Do we need the line formatting here ? We could at least have a 80 chars policy. This one ends at 66.